pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -394,6 +394,10 @@ func (gce *GCECloud) EnsureTCPLoadBalancer(name, region string, externalIP net.I
 		PortRange:  fmt.Sprintf("%d-%d", minPort, maxPort),
 		Target:     gce.targetPoolURL(name, region),
 	}
+	if loadBalancerIP != nil {
+		req.IPAddress = loadBalancerIP.String()
```

Comment:
Where does it get the IP from today?

On Mon, Aug 31, 2015 at 11:14 PM, Angus Lees <notifications@github.com>
wrote:

> In pkg/cloudprovider/providers/gce/gce.go
> <https://github.com/kubernetes/kubernetes/pull/13005#discussion_r38388282>
> :
>
> > @@ -394,6 +394,10 @@ func (gce *GCECloud) EnsureTCPLoadBalancer(name, region string, externalIP net.I
> >  		PortRange:  fmt.Sprintf("%d-%d", minPort, maxPort),
> >  		Target:     gce.targetPoolURL(name, region),
> >  	}
> > +	if loadBalancerIP != nil {
> > +		req.IPAddress = loadBalancerIP.String()
>
> Aha, thanks - I think I finally understand now. The OpenStack plugin has
> always (perhaps mistakenly) implemented externalIP exactly as you're trying
> to do with this change, hence my confusion. On the upside, that means the
> plugin should just continue to work once we formalise that interpretation :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13005/files#r38388282>.
>